src/Symfony/Component/Cache/Adapter/AbstractAdapter.php

```diff
@@ -169,7 +199,13 @@ public function hasItem($key)
             $this->commit();
         }
 
-        return $this->doHave($this->getId($key));
+        $got = $this->doHave($this->getId($key));
+
+        if ($got || !$this->decorated) {
```

Comment:
In the other places we have this condition the other way around (checking if we did not find anything and then calling the decorated adapter). For consistency, I suggest to do the same here too.